protocol/contracts/templegold/TempleGold.sol

```diff
@@ -0,0 +1,331 @@
+    /// @notice 1B max supply
+    uint256 public constant MAX_SUPPLY = 1_000_000_000 ether; // 1B
+    /// @notice Minimum Temple Gold minted per call to mint
+    uint256 public constant MINIMUM_MINT = 1_000;
+
+    /// @notice Mint chain id
+    uint256 private immutable mintChainId;
+
+    /// @notice Whitelisted addresses for transferrability
+    mapping(address => bool) public override authorized;
+    /// @notice Distribution parameters. Minted share percentages for staking, escrow and gnosis. Adds up to 100%
+    DistributionParams private distributionParams;
+    /// @notice Vesting factor determines rate of mint
+    VestingFactor private vestingFactor;
+
+    constructor(
+        InitArgs memory _initArgs
+    ) OFT(_initArgs.name, _initArgs.symbol, _initArgs.layerZeroEndpoint, _initArgs.executor) Ownable(_initArgs.executor){
+       staking = ITempleGoldStaking(_initArgs.staking);
+       escrow = IDaiGoldAuction(_initArgs.escrow);
+       teamGnosis = _initArgs.gnosis;
+       mintChainId = _initArgs.mintChainId;
+    }
+
+    /**
+     * @notice Set staking proxy contract address
+     * @param _staking Staking proxy contract
+     */
+    function setStaking(address _staking) external override onlyOwner {
+        if (_staking == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        staking = ITempleGoldStaking(_staking);
+        emit StakingSet(_staking);
+    }
+
+    /**
+     * @notice Set auctions escrow contract address
+     * @param _escrow Auctions escrow contract address
+     */
+    function setEscrow(address _escrow) external override onlyOwner {
+        if (_escrow == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        escrow = IDaiGoldAuction(_escrow);
+        emit EscrowSet(_escrow);
+    }
+
+    /**
+     * @notice Set team gnosis address
+     * @param _gnosis Team gnosis address
+     */
+    function setTeamGnosis(address _gnosis) external override onlyOwner {
+        if (_gnosis == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        teamGnosis = _gnosis;
+        emit TeamGnosisSet(_gnosis);
+    }
+
+    /**
+     * @notice Whitelist an address to allow transfer of Temple Gold to or from
+     * @param _contract Contract address to whitelist
+     * @param _whitelist Boolean whitelist state
+     */
+    function authorizeContract(address _contract, bool _whitelist) external override onlyOwner {
+        if (_contract == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        authorized[_contract] = _whitelist;
+        emit ContractAuthorizationSet(_contract, _whitelist);
+    } 
+
+    /**
+     * @notice Set distribution percentages of newly minted Temple Gold
+     * @param _params Distribution parameters
+     */
+    function setDistributionParams(DistributionParams calldata _params) external override onlyOwner {
+        if (_params.staking < MINIMUM_DISTRIBUTION_SHARE 
```

Comment:
So you mean setting team gnosis share to 0% temporarily? My understanding is gnosis share is a fixed percentage of supply. I will revert to logicians to be sure

Reply:
For example, in case we want to say that team doesn't get any supply for a few epochs or whatever for sales/marketing/etc.

Or if we want to stop staking from getting supply for whatever reason, etc.